### PR TITLE
test on 3.10

### PR DESCRIPTION
**Pull request by graingert**
/cc @Kludex 
 
(running with warnings as error enabled)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,9 +10,9 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (graingert) [inline .github/workflows/test.yml:13]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9, 3.10.0-beta.4]
```

**Comment by Kludex** [comment]
This branch was created on top of #2298 (just for reference).

I guess @tiangolo will want to merge them separately, so we can have two different commits on `master`: "Support 3.9" and "Support 3.10".

File: pyproject.toml [hunk 2]
```diff
@@ -28,6 +28,7 @@ classifiers = [
     "Programming Language :: Python :: 3.6",
     "Programming Language :: Python :: 3.7",
     "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
```
Comment (graingert) [inline pyproject.toml:31]: I'm not going to add the 3.10 trove classifier until https://github.com/samuelcolvin/pydantic/pull/2885 is released

/cc @PrettyWood

File: pyproject.toml [hunk 3]
```diff
@@ -110,6 +111,7 @@ junit_family = "xunit2"
 filterwarnings = [
     "error",
     'ignore:"@coroutine" decorator is deprecated since Python 3\.8, use "async def" instead:DeprecationWarning',
```
Comment (graingert) [inline pyproject.toml:113]: they threatened to remove this in 3.10 but it's actually gone in 3.11

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.6, 3.7, 3.8, 3.9, 3.10.0-beta.4]
```
Comment (graingert) [inline .github/workflows/test.yml:13]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9, 3.10.0-rc.2]
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.6, 3.7, 3.8, 3.9, 3.10.0-rc.2]
```
Comment (graingert) [inline .github/workflows/test.yml:15]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9, "3.10"]
```

File: pyproject.toml [hunk 6]
```diff
@@ -140,6 +141,7 @@ junit_family = "xunit2"
...
+    "ignore:There is no current event loop:DeprecationWarning",
```
Comment (graingert) [inline pyproject.toml:144]:
```suggestion
```

**Comment by graingert** [comment]
closing in favour of https://github.com/tiangolo/fastapi/pull/3712/files

**Pull request closed without merging**